### ISSUE TITLE
TileGrid shifts XYZ-Basemaps about 5km

### ISSUE THREAD

**Issue by JakobMiksch** [meggsimum/wegue#166]
It seems like the TileGrid change causes a shift of XYZ-Basemaps (~5km). The "standard" OSM Basemap via the custom OL-source is not affected. See GIF below.  Below also the layer definition in  `app-config.json`
When I remove the `tileGrid` Property in the `LayerFactory`, everything works as expected.
@justb4 

```javascript
  /**
   * Returns an XYZ based tile layer instance due to given config.
   *
   * @param  {Object} lConf  Layer config object
   * @return {ol.layer.Tile} OL XYZ layer instance
   */
  createXyzLayer (lConf) {
    const xyzLayer = new TileLayer({
      name: lConf.name,
      lid: lConf.lid,
      displayInLayerList: lConf.displayInLayerList,
      visible: lConf.visible,
      opacity: lConf.opacity,
      source: new XyzSource({
        url: lConf.url,
        attributions: lConf.attributions,
      //  tileGrid: lConf.tileGrid,  // <-- this probably causes the shift
        projection: lConf.projection
      })
    });
```

![Peek 2021-01-28 21-01](https://user-images.githubusercontent.com/15704517/106192163-19cbfc00-61ac-11eb-8401-db036b976e23.gif)

Layer Definition of `app-config.json`:
```json
    {
      "type": "VECTOR",
      "lid": "gewerbe",
      "name": "Gewerbe Mutterstadt",
      "url": "./static/data/gewerbe.geojson",
      "format": "GeoJSON",
      "visible": true,
      "selectable": true,
      "style": {
        "radius": 4,
        "strokeColor": "purple",
        "strokeWidth": 2,
        "fillColor": "rgba(155,153,51,0.5)",
        "label": {
          "attribute": "title",
          "minResolution": 4.0,
          "outlineColor": "white",
          "outlineWidth": 2,
          "fillColor": "black",
          "offsetX": 0,
          "offsetY": 15,
          "align": "center"
        }
      }
    },
    {
      "type": "XYZ",
      "name": "Carto Positron",
      "url": "http://basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png",
      "lid": "positron",
      "displayInLayerList": true,
      "visible": true,
      "attributions": "© <a href='https://carto.com/attribution/' target='_blank'>CARTO</a> | <a href='https://openstreetmap.org/copyright' target='_blank'>© OpenStreetMap</a>-contributors | "
    }
```

**Comment by chrismayer** [comment]
I just discovered that this (or at least similar unwanted shift) appeared for a WMS layer in case an explicit `projection` property is set:

```json
    {
      "type": "WMS",
      "lid": "terrestris-bw-wms",
      "name": "OSM WMS",
      "format": "image/png",
      "layers": "OSM-WMS",
      "url": "https://ows.terrestris.de/osm-gray/service",
      "transparent": true,
      "singleTile": false,
      "attributions": "© <a href='https://terrestris.de' target='_blank'>terrestris</a> (<a href='https://www.openstreetmap.org/copyright' target='_blank'>© OpenStreetMap-Mitwirkende</a>) |",
      "isBaseLayer": false,
      "visible": true,
      "displayInLayerList": true,
      "projection": "EPSG:3857"
    }
``` 

By removing ` "projection": "EPSG:3857"`  (which is the default) ensured that there is no map shift.

**Comment by JakobMiksch** [comment]
I added the layers definition of the `app-config.json`

**Comment by chrismayer** [comment]
@JakobMiksch I provided a fix for this (#170). Could you test it, if it solves your initial problem?

**Comment by JakobMiksch** [comment]
@chrismayer yes, it does fix the problem  :+1:
